### ISSUE TITLE
Unmapped track - Australia, Mount Kaye Pass

### ISSUE THREAD

**Issue by fenugrec** [soong-construction/dirt-rally-time-recorder#60]
Weird, the length is close but not identical to the entry in the db:

```
2021-12-22 16:51:22,866 DEBUG [timerecorder.respawnTracker] RESTART
2021-12-22 16:51:27,314 WARNING [timerecorder.databaseAccess] Failed to identify track
2021-12-22 16:51:27,315 DEBUG [timerecorder.databaseAccess] Length: 13304.1201171875 (Z: 2242.289306640625)
2021-12-22 16:51:27,315 INFO [timerecorder.statsProcessor] CAR: Volkswagen Golf GTI 16V
2021-12-22 16:51:27,315 DEBUG [timerecorder.statsProcessor] q1614560583.-1.200.started
2021-12-22 16:51:27,316 INFO [timerecorder.statsProcessor] Volkswagen Golf GTI 16V: H-PATTERN shifting, 5 speed, with manual CLUTCH, with HANDBRAKE
2021-12-22 17:00:03,720 ERROR [timerecorder.ambiguousResultHandler] Unmapped track telemetry. Please report at https://github.com/soong-construction/dirt-rally-time-recorder
2021-12-22 17:00:03,720 DEBUG [timerecorder.ambiguousResultHandler] UPDATE laptimes SET Track=??? WHERE Timestamp="1640210403.720245";
2021-12-22 17:00:03,726 DEBUG [timerecorder.statsProcessor] q1614560583.-1.200.time:511.32|s
2021-12-22 17:00:03,726 DEBUG [timerecorder.statsProcessor] q1614560583.-1.200.topspeed:158.4|kph
2021-12-22 17:00:03,726 INFO [timerecorder.statsProcessor] Completed stage in 08:31.320.

```
db has `13301.109375 | -2352.93017578125`

wonder if they changed something in recent updates ? not clear

**Comment by maxbechtold** [comment]
Hi fenugrec, appreciate the report! 
This is unexpected, the track/stage length should be a pretty clear indicator - could be Codemasters changed that, or perhaps the DRTR mapping was wrong from the start.
Do you recall what mode it was - time trial, online event, single player event?

If you encounter any other mismatching tracks, feel free to report. I'll see if I can check up on it during the holidays.

**Comment by maxbechtold** [comment]
This must been a mapping error, as the length of Mount Kaye Pass and Mount Kaye Pass Reverse are identical and DRTR cannot map the first, but finds two mappings for the latter:
```
2021-12-23 14:56:27,803 DEBUG [timerecorder.inputTracker] LAUNCHED <--- *Mount Kaye Pass*
2021-12-23 14:58:51,397 DEBUG [timerecorder.respawnTracker] RECOVER
2021-12-23 14:58:51,398 WARNING [timerecorder.databaseAccess] Failed to identify track
2021-12-23 14:58:51,398 DEBUG [timerecorder.databaseAccess] Length: 13304.1201171875 (Z: 2242.09619140625)
2021-12-23 14:58:51,398 VERBOSE [timerecorder.databaseAccess] INSERT INTO Tracks (id, name, length, startz) VALUES (ID, 'TRACK_NAME', 13304.1201171875, 2242.09619140625);
2021-12-23 14:58:51,398 INFO [timerecorder.statsProcessor] CAR: Volkswagen Golf Kitcar
2021-12-23 14:58:51,398 DEBUG [timerecorder.statsProcessor] Max1640267505.-1.1302.started
2021-12-23 14:58:51,398 INFO [timerecorder.statsProcessor] Volkswagen Golf Kitcar: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
2021-12-23 14:59:42,897 DEBUG [timerecorder.respawnTracker] RESTART <--- *Mount Kaye Pass Reverse*
2021-12-23 14:59:42,911 WARNING [timerecorder.databaseAccess] Ambiguous track data, 2 matches
2021-12-23 14:59:42,911 DEBUG [timerecorder.databaseAccess] Length: 13301.109375 (Z: -2353.145751953125)
2021-12-23 14:59:42,912 INFO [timerecorder.statsProcessor] CAR: Volkswagen Golf Kitcar
2021-12-23 14:59:42,912 DEBUG [timerecorder.statsProcessor] Max1640267505.-1.1302.started
2021-12-23 14:59:42,912 INFO [timerecorder.statsProcessor] Volkswagen Golf Kitcar: SEQUENTIAL shifting, 6 speed, with HANDBRAKE
``` 
There is another such pair of tracks, it should be checked if these are properly recognized, too:
`Col de Turini - Descente|Gordolon - Courte montee`

**Comment by fenugrec** [comment]
>  Do you recall what mode it was - time trial, online event, single player event?
It was time trial. I hope they don't change the tracks in subtle ways between modes ...

I can check that other pair later today maybe. Although those are in monte carlo... I avoid that location as much as possible P )

**Comment by fenugrec** [comment]
Gordolon - courte montée : OK
COl de Turini -descente : OK

**Comment by maxbechtold** [comment]
Thanks for checking, I don't have DR2 installed on this machine to verify it myself right now. Those two Monaco stages can be told apart by the different altitude of the stage start (z-parameter).

I noticed the stage name was wrong as well... 

I might tag the next release version soon. As I recall you build DRTR yourself, so I added a tag fix-issue60 that fixes the mapping if you need it earlier.

**Comment by maxbechtold** [comment]
Resolved with version 2.10.5, therefore closing.
